### PR TITLE
Update faucet API to support batching, and separate into it's own export

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
## Description 

This is a replacement for #12668, updating all of the faucet APIs to a new export so that it can be used on it's own. The provider and signer methods are now marked as deprecated, and will be removed in a subsequent release.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by Jibz-Mysten — APPROVED** [review]

**Review by healthydeve — APPROVED** [review]

File: sdk/typescript/package.json [hunk 1]
```diff
@@ -42,6 +42,11 @@
 			"source": "./src/keypairs/secp256r1/index.ts",
 			"import": "./dist/esm/keypairs/secp256r1/index.js",
 			"require": "./dist/cjs/keypairs/secp256r1/index.js"
+		},
```
Comment (healthydeve) [inline sdk/typescript/package.json:45]: Noting to myself that this is how you add a new package

**Review by plam-ml — APPROVED** [review]

File: sdk/typescript/package.json (hunk 1 above)
Comment (Jordan-Mysten) [reply sdk/typescript/package.json:45]: This isn’t a package it’s just a way to more granularly import out of the package. It’s an approach we’re trying to take to minimize the required size of the SDK.

File: sdk/typescript/src/faucet/index.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { ObjectId, SuiAddress, TransactionDigest } from '../types/index.js';
+import { FaucetRateLimitError } from '../utils/errors.js';
```
Comment (hayes) [inline sdk/typescript/src/faucet/index.ts:5]: This could probably move into the faucet directory

File: sdk/typescript/faucet/package.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (hayes) [inline sdk/typescript/faucet/package.json:1]: You'll need to add this to the files section of the root package.json or it won't make it into the package when it's published

**Review by hayes-mysten — APPROVED** [review]

**Pull request merged**